### PR TITLE
Morse code

### PR DESCRIPTION
**Pull request by sarastavnitser**

### CODE REVIEW

**Review by sarastavnitser — COMMENTED** [review]
not sure why my mockito for the presenter isnt working

File: src/main/java/MorseCodeConverter.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        String[] morses = {"a .-", "b -...", "c -.-.", "d -..", "e .", "f ..-.", "g --.", "h ....", "i ..", "j .---", "k -.-",
+                "l .-..", "m --", "n -.", "o ---", "p .--.", "q --.-", "r .-.", "s ...", "t -", "u ..-", "v ...-", "w .--", "x -..-",
+                "y -.--", "z --..", "0 -----", "1 .----", "2 ..---", "3 ...--", "4 ....-", "5 .....", "6 -....", "7 --...", "8 ---..",
+                "9 ----."};
+        for (int i = 0; i < morses.length; i++) {
+            String defSplit[] = morses[i].split(" ");
+            morse.put(defSplit[0], defSplit[1]);
+            revMorse.put(defSplit[1], defSplit[0]);
+            morse.put(" ", " ");
+            revMorse.put(" ", " ");
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java:19]: Why?

File: src/main/java/MorseCodeConverter.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+    }
+
+
+    String toMorse(String inp) {
+        inp = inp.toLowerCase();
+        StringBuilder sb = new StringBuilder();
+        char[] inpChar = inp.toCharArray();
+        for (char letter : inpChar) {
+            if (morse.get(letter + "") != null) {
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java:31]: Don't call `morse.get()` twice.
Also don't use `+ ""`, use `String.valueOf(letter)` instead.

File: src/main/java/MorseCodeConverter.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        return sb.toString().trim();
+    }
+
+    String toEnglish(String inp) {
+        StringBuilder sb = new StringBuilder();
+        String elements[] = inp.split(" ");
+        for (int i = 0; i < elements.length; i++) {
+            if (revMorse.get(elements[i] + "") != null) {
+                sb.append(revMorse.get(elements[i] + ""));
```
Comment (andrewoid) [inline src/main/java/MorseCodeConverter.java:49]: same as above

**Comment by andrewoid** [comment]
You need the add the yaml file so that your PR runs your tests.

File: src/main/java/MorseCodeConverter.java (hunk 1 above)
Comment (sarastavnitser) [reply src/main/java/MorseCodeConverter.java:19]: I put it in for if the user enters spaces, should I take it out and just have spaces under not compatible with morse code?

**Review by andrewoid — APPROVED** [review]

**Pull request merged**